lib/tapyrus/script/script.rb

```diff
@@ -40,6 +40,33 @@ def to_p2sh
       Script.to_p2sh(to_hash160)
     end
 
+    # generate cp2pkh script
+    # @param [ColorIdentifier] color identifier
+    # @param [String] hash160 of pubkey
+    # @return [Script] CP2PKH script
+    def self.to_cp2pkh(color_id, pubkey_hash)
+      new << color_id.to_payload << OP_COLOR << OP_DUP << OP_HASH160 << pubkey_hash << OP_EQUALVERIFY << OP_CHECKSIG
```

Comment:
It seems to me that when building or checking cp2pkh and cp2sh, it should check the validity of ColorIdentifier. What do you think?

Reply:
Agree. 
add validation acedb68734b67110d1ed57922bd4873aee841473